benchmarks/Surrogates/tensor_product.jmd

```diff
@@ -0,0 +1,147 @@
+---
+title: Tensor Product Function
+author: Mridul Jain, Sathvik Bhagavan, Chris Rackauckas
+---
+
+The tensor product function is defined as:
+
+``f(x) =  \prod_{i=1}^{d} \cos(a\pi x_i)``
+
+where:
+
+- \(d\): Represents the dimensionality of the input vector \(x\).
+- \(x_i\): Represents the \(i\)-th component of the input vector.
+- \(a\): A constant parameter.
+
+## Package Imports
+
+```julia
+using Surrogates
+using SurrogatesRandomForest
+using Plots
+using Statistics
+using PrettyTables
+using BenchmarkTools
+default()

```

Comment:
what's this?

Reply:
I have used ``PrettyTables`` for tabular display of training mse , testing mse and times of the surrogate models and ``BenchmarkTools`` for timing the models.

Reply:
Chris, are you referring to `default()`? That can be removed, I missed that.

Reply:
But what is `default`?

Reply:
Give me a sec, I am removing ``default()``

Reply:
It is a function in Plots.jl to set defaults:

```julia
help?> Plots.default()
  default(key) returns the current default value for that key.

  default(key, value) sets the current default value for that key.

  default(; kw...) will set the current default value for each key/value pair.

  default(plotattributes, key) returns the key from plotattributes if it exists, otherwise default(key).
```

Reply:
That's not necessary though. It can just be removed.

Reply:
yes

Reply:
@ChrisRackauckas  @sathvikbhagavan 
So I should remove it then right?

Reply:
yes

Reply:
yes